chromadb/test/db/test_log_purge.py

```diff
@@ -24,3 +24,27 @@ def test_log_purge(sqlite_persistent: System) -> None:
 
     # The purge of the second collection should not be blocked by the first
     invariants.log_size_below_max(client._system, collections, True)
+
+
+def test_log_purge_with_multiple_collections(sqlite_persistent: System) -> None:
+    client = Client.from_system(sqlite_persistent)
+
+    first_collection = client.create_collection(
+        "first_collection", metadata={"hnsw:sync_threshold": 10, "hnsw:batch_size": 10}
+    )
+    second_collection = client.create_collection(
+        "second_collection", metadata={"hnsw:sync_threshold": 10, "hnsw:batch_size": 10}
+    )
+    collections = [first_collection, second_collection]
+
+    # (Does not trigger a purge)
+    for i in range(15):
+        first_collection.add(ids=str(i), embeddings=[i, i])
+
+    # (Should trigger a purge)
+    for i in range(25):
+        second_collection.add(ids=str(i), embeddings=[i, i])
+
+    invariants.log_size_for_collections_match_expected(
+        client._system, collections, True
+    )
```

Comment:
should we just run our normal invariants here? (all embeddings exist, querying works, etc)

Reply:
I thought to be on the safe-side I'd add an explicit check as the expected number of entries in the WAL are known (count%batch_size+1)

Reply:
As discussed on Discord, running invariants in the current test won't catch the defect as the BruteForce holds reference to the embeddings. The latter are lost only when the collection is unloaded/loaded again.

A good test, we discussed to have is to ensure the durability of things - aka `test_persist` where after initial ingestion the collection invariants are checked in a separate process.